org.eclipse.jdt.ls.core/src/org/eclipse/jdt/ls/core/internal/contentassist/CompletionProposalRequestor.java

```diff
@@ -337,6 +347,32 @@ public CompletionItem toCompletionItem(CompletionProposal proposal, int index) {
 			if (replace != null && replace.getEnd().getLine() != replace.getStart().getLine()) {
 				replace.setEnd(replace.getStart());
 			}
+			if (preferenceManager.getClientPreferences().isCompletionListItemDefaultsEditRangeSupport()) {
```

Comment:
I'd avoid placing the one-time initialization logic to be done on every `toCompletionItem(..)`. This is called only by `getCompletionItems(IProgressMonitor)`. I would put it somewhere there. You could even create some helper method, `initializeCompletionListDefaults(..)` that just takes the first item and sets things up.

You'll still have some logic here comparing the default to the existing values of the proposal to determine if we need to set non-default values.